cms/blog/models.py

```diff
@@ -0,0 +1,125 @@
+from django import forms
+from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
+from django.db import models
+from modelcluster.fields import ParentalManyToManyField
+from wagtail.wagtailadmin.edit_handlers import (FieldPanel, ObjectList,
+                                                PageChooserPanel,
+                                                TabbedInterface)
+from wagtail.wagtailcore.fields import RichTextField
+from wagtail.wagtailcore.models import Page
+from wagtail.wagtailimages.edit_handlers import ImageChooserPanel
+
+from cms.snippets.models import Category
+
+
+class BlogIndexPage(Page):
+
+    title_blog_index = models.CharField(
+        max_length=255, blank=True, verbose_name="Blog Index Title")
+    featured_page = models.ForeignKey(
+        'cms_blog.BlogPage',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+',
+    )
+
+    subpage_types = ['cms_blog.BlogPage']
+
+    @property
+    def blogs(self):
+        blogs = BlogPage.objects.live()
+        blogs = blogs.order_by('-create_date')
+        return blogs
+
+    def get_context(self, request):
+        blogs = self.blogs
+
+        category = request.GET.get('category')
+        if category:
+            blogs = blogs.filter(categories__name_en=category)
+
+        page = request.GET.get('page')
+        paginator = Paginator(blogs, 5)
+        try:
+            blogs = paginator.page(page)
+        except PageNotAnInteger:
+            blogs = paginator.page(1)
+        except EmptyPage:
+            blogs = paginator.page(paginator.num_pages)
+
+        # Only show five page numbers in pagination
```

Comment:
Nice. In the long term, we can maybe factor that logic out and reuse it for pagination on the projects overview as well.